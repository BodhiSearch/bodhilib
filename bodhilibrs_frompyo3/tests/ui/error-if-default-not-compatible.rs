use frompyo3::FromPyO3;

#[derive(FromPyO3)]
struct TestDefault {
  #[frompyo3(default = 1)]
  recursive: bool,
}

fn main() {}
