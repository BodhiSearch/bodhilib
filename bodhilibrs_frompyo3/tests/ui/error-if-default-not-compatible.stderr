error[E0308]: `if` and `else` have incompatible types
 --> tests/ui/error-if-default-not-compatible.rs:3:10
  |
3 | #[derive(FromPyO3)]
  |          ^^^^^^^^
  |          |
  |          expected `Result<{integer}, _>`, found `Result<bool, _>`
  |          expected because of this
  |
  = note: expected enum `Result<{integer}, _>`
             found enum `Result<bool, _>`
  = note: this error originates in the derive macro `FromPyO3` (in Nightly builds, run with -Z macro-backtrace for more info)
